packages/expo/src/logs/LogSerialization.ts

```diff
@@ -73,7 +73,17 @@ function _stringifyLogData(data: unknown[]): string[] {
     if (typeof item === 'string') {
       return item;
     } else {
-      return prettyFormat(item, { plugins: [ReactNodeFormatter] });
+      //check the size of string returned
+      let result = prettyFormat(item, { plugins: [ReactNodeFormatter] });
+      if(result.length > 10000){
+        //truncate the result to 10kb
+        let newResult = result.substring(0,10000);
+        newResult += '...';
```

Comment:
Communicate clearly what is happening with the truncation when people see this in their logs:
```suggestion
        newResult += `...(truncated to the first ${LOG_MESSAGE_MAX_LENGTH} characters)`;
```